src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -910,5 +943,42 @@ private ICollection<ProjectWithInnerNodes> LoadProjects(IEnumerable<ProjectGraph
             // Just return the projects not the whole dictionary as it was just used to group the projects together
             return projects.Values;
         }
+
+        /// <summary>
+        /// It evaluates the project and returns true if the project has CentralPackageVersionManagement enabled.
+        /// </summary>
+        /// <param name="project">The <see cref="IMSBuildProject"/> for which the CentralPackageVersionManagement will be evaluated.</param>
+        /// <param name="projectStyle">The <see cref="ProjectStyle?"/>. Null is the project did not have a defined ProjectRestoreStyle property.</param>
+        /// <returns>True if the project has CentralPackageVersionManagement enabled and the project is PackageReference or the projectStyle is null.</returns>
+        internal static bool IsCentralVersionsManagementEnabled(IMSBuildProject project, ProjectStyle? projectStyle)
+        {
+            var cpvmEnabledProperty = StringComparer.OrdinalIgnoreCase.Equals(project.GetProperty("CentralPackageVersionsEnabled"), bool.TrueString);
+
+            return (cpvmEnabledProperty && (!projectStyle.HasValue || (projectStyle.Value == ProjectStyle.PackageReference)));
+        }      
+
+        /// <summary>
+        /// It evaluates the <paramref name="item"/> metadata and search for the <paramref name="propertyName"/> metadata key.
+        /// If the metadata exists and has the value "true" the method will return <see cref="true"/>, otherwise it wil return <see cref="false"/>.
+        /// </summary>
+        /// <param name="item">The item to be evaluated.</param>
+        /// <param name="propertyName">The property name.</param>
+        /// <returns>Returns true only if the metadata with the <paramref name="propertyName"/> key exists and its value is <see cref="true"/>.</returns>
+        private static bool IsPropertyTrue(IMSBuildItem item, string propertyName)
+        {
+            return StringComparer.OrdinalIgnoreCase.Equals(item.GetProperty(propertyName), bool.TrueString);
+        }
```

Comment:
Could re-use this helper method: https://github.com/NuGet/NuGet.Client/blob/66a01b3c4df37bcf567b072443629b558241a7bd/src/NuGet.Core/NuGet.Common/MsBuildStringUtility.cs#L74-L77

Reply:
Removed.